runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -497,25 +535,20 @@ private void checkTimestamp(Instant timestamp) {
             String name, CombineFn<AggregatorInputT, ?, AggregatorOutputT> combiner) {
       return context.createAggregator(name, combiner);
     }
-  }
-
-  private class DoFnExtraContextFactory<InputT, OutputT>
-      implements DoFn.ExtraContextFactory<InputT, OutputT> {
-
-    /** The windows of the current element. */
-    private final Collection<? extends BoundedWindow> windows;
 
-    /** The pane of the current element. */
-    private final PaneInfo pane;
+    @Override
+    public BoundedWindow window() {
+      return Iterables.getOnlyElement(windowedValue.getWindows());
+    }
 
-    public DoFnExtraContextFactory(Collection<? extends BoundedWindow> windows, PaneInfo pane) {
-      this.windows = windows;
-      this.pane = pane;
+    @Override
+    public DoFn<InputT, OutputT>.Context context(DoFn<InputT, OutputT> doFn) {
+      return this;
```

Comment:
Agree. I noticed that we do this here and there. But I think as of this PR and a couple others we can steer away from both of those parameters.